pandas/core/internals/array_manager.py

```diff
@@ -684,7 +684,10 @@ def get_numeric_data(self, copy: bool = False) -> ArrayManager:
         copy : bool, default False
             Whether to copy the blocks
         """
-        return self._get_data_subset(lambda arr: is_numeric_dtype(arr.dtype))
+        return self._get_data_subset(
+            lambda arr: is_numeric_dtype(arr.dtype)
+            or getattr(arr.dtype, "_is_numeric", False)
```

Comment:
seems like is_numeric_dtype should catch this?

Reply:
It maybe should, but it doesn't at the moment. 

The above is mimicking the logic of the BlockManager (ExtensionBlock.is_numeric checks this attribute of the dtype). But it's certainly inconsistent .. The consequence of the inconsistency is that eg decimal is considered numeric for groupby operations, but `is_numeric_dtype(decimal_dtype)` will return False (not sure where this would have a direct impact).